fix invalid peer dependency

timaschew opened the pull request:
I get this error when try to run `npm i`
```
npm WARN unmet dependency /Users/awilhelm/dev/redux-router/node_modules/react-dom requires react@'^0.14.0' but will load
npm WARN unmet dependency /Users/awilhelm/dev/redux-router/node_modules/react,
npm WARN unmet dependency which is version 0.14.0-rc1
npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/bin/iojs" "/usr/local/bin/npm" "i"
npm ERR! node v4.1.2
npm ERR! npm  v2.14.4
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package react@0.14.0-rc1 does not satisfy its siblings' peerDependencies requirements!
```

btw: how I can run the basic example? there is no  webpack command to bundle the files. and how the files are served? After running `npm build` I still cannot run any server.js with node, the process is stopping immediately without any errors.

gaearon commented:
>btw: how I can run the basic example? there is no webpack command to bundle the files. and how the files are served? After running npm build I still cannot run any server.js with node, the process is stopping immediately without any errors.

```
npm install
cd examples/basic
npm install
npm start
```

Does this help?

timaschew commented:
Oh, I didn't see that there is a npm script, it's a little bit confusing, because there are 3 package.json files.

this is how it worked, first fix the invalid peer dependencies in the root package.json then
```
npm i
cd examples
npm i
cd basic
npm start
```

I added a `npm run basic` in the package.json of the `examples` directory
so now it's

```
npm i
cd examples
npm i
npm run basic
```
Then there was an error, because the alias for redux-router was wrong (old name).
And then I fixed some warnings with the react-transform.

There are still 2 warnings in the browser, the one is about the devtools (I read you're on it) the other is about the `key` prop. I tried to write this
`<Link key={l} to={l}>{l}</Link>` but the warning was still there.

Scarysize commented:
Hey, if you update to be compliant with the latest version, I will merge it :+1:
